src/main/java/com/spartronics4915/frc2020/subsystems/LED.java

```diff
@@ -0,0 +1,100 @@
+package com.spartronics4915.frc2020.subsystems;
+
+import com.spartronics4915.lib.util.Logger;
+import com.spartronics4915.lib.subsystems.SpartronicsSubsystem;
+
+import edu.wpi.first.wpilibj.SerialPort;
+import edu.wpi.first.wpilibj.smartdashboard.SmartDashboard;
+
+/**
+ * The subsystem that controls the LED.
+ * Communicating to the Arduino by giving
+ * a byte using Serial to the wanted BlingState to be
+ * read by the Arduino to show the given Bling style wanted.
+ *
+ * Based off the Arduino code under Spartronics4915/Bling/Roborio_functionality
+ */
+public class LED extends SpartronicsSubsystem
+{
+    private SerialPort mBling;
+
+    private static LED sInstance = null;
+    private static BlingState mBlingState;
+
+    public static LED getInstance()
+    {
+        if (sInstance == null)
+        {
+            sInstance = new LED();
+            mBlingState = BlingState.OFF;
+        }
+        return sInstance;
+    }
+
+    private LED()
+    {
+        try
+        {
+            mBling = new SerialPort(9600, SerialPort.Port.kUSB);
+
+            logInitialized(true);
+        }
+        catch (Exception e)
+        {
+            logException("LED: Couldn't initialize SerialPort", e);
+            logInitialized(false);
+        }
+    }
+
+    // TODO: validate desired periodic() function for LED subsystem
+    @Override
+    public void periodic()
+    {
+        SmartDashboard.putString("LED state:", mBlingState.toString());
+    }
+
+    /**
+     * This enum is giving the possible styles we can have the Arduino express.
+     */
+    public enum BlingState
+    {
+        // TODO: Add blingStates & ensure it matches Arduino sketch code
+        // bling code is passed in for use in the BlingState methods
+        OFF("0"), // turn off bling
+        DISABLED("1"), // robot powered on but disabled (in disabledInit())
+        SOME_BLING_STATE_2("2"), // ...
+        SOME_BLING_STATE_3("3"), // ...
+        ; // semicolon to state more to follow
+
+        private final String blingCode;
+
+        BlingState(String code)
+        {
+            this.blingCode = code;
+        }
+
+        public byte[] getBlingMessage()
+        {
+            return this.blingCode.getBytes();
+        }
+    }
+
+    /**
+     * This will go through what we want the bling to do and express that style of bling.
+     */
+    public void setBlingState(BlingState blingState)
+    {
+        if (!isInitialized())
+        {
+            return;
+        }
+
+        // save current blingState for smartdashboard display
+        mBlingState = blingState;
+
+        // convert state to byte message and sent to serial port
+        byte[] message = blingState.getBlingMessage();
+        mBling.write(message, message.length);
```

Comment:
I bet this call could fail, such as if the Arduino became disconnected during a match. We should check the return code or serial port status in some way, and 'de-initialize' this module if we get an error.

Reply:
In other words, before the 'return;' we could log:
```
Logger.notice("LED: (UNINITIALIZED) setBlingState is set to: " + blingState.name());
```
And then have a regular log entry after the check.

Reply:
adding todo